spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/ConsulHeartbeatAutoConfiguration.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.consul;
```

Comment:
Why the move up a package?

Reply:
The TTL scheduler is used by both the discovery as the service registry.
Made sense to move it up because of [this comment](https://github.com/spring-cloud/spring-cloud-consul/pull/574/files/5a889e4ce39e833bd45d44bc68424965caffb926#diff-acf0e7c3a3b38764a1361ebcbf0e596eR61)

Reply:
ok, let's move it to a separate sub-package then.

Reply:
Now I'm not sure anymore because of the conditional `spring.cloud.consul.discovery.heartbeat.enabled` property 😕

Reply:
no problems for me. put it in a `support` package. Property doesn't have to match package.